src/pudl/transform/vceregen.py

```diff
@@ -0,0 +1,206 @@
+"""Transformations of the Vibrant Clean Energy renewable generation profiles.
+
+Wind and solar profiles are extracted separately, but concatenated into a single table
+in this module, as they have exactly the same structure.
+"""
+
+import pandas as pd
+from dagster import asset
+
+import pudl
+from pudl.helpers import cleanstrings_snake, simplify_columns, zero_pad_numeric_string
+from pudl.metadata.dfs import POLITICAL_SUBDIVISIONS
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+def _prep_lat_long_fips_df(raw_vcegen__lat_lon_fips: pd.DataFrame) -> pd.DataFrame:
+    """Prep the lat_long_fips table to merge into the capacity factor tables.
+
+    Prep entails making sure the formatting and column names match those in the
+    capacity factor tables, adding 0s to the beginning of FIPS codes with 4 values,
+    and making separate county and state columns.
+    """
+    logger.info(
+        "Preping Lat-Long-FIPS table for merging with the capacity factor tables"
+    )
+    # Get state name patterns from the POLITICAL_SUBDIVISIONS table
+    ps_usa_df = POLITICAL_SUBDIVISIONS[POLITICAL_SUBDIVISIONS["country_code"] == "USA"]
+    state_names = cleanstrings_snake(
+        ps_usa_df, ["subdivision_name"]
+    ).subdivision_name.tolist()
+    state_pattern = "|".join(state_names)
+    # Transform lat long fips table by making the county_state_names lowercase
+    # to match the values in the capacity factor tables. Fix FIPS codes with
+    # no leading zeros, and add a unique country and state field.
+    lat_long_fips = (
+        raw_vcegen__lat_lon_fips.pipe(simplify_columns)
+        .assign(county_state_names=lambda x: x.county_state_names.str.lower())
+        .assign(fips=lambda x: zero_pad_numeric_string(x.fips, 5))
+        .assign(
+            county=lambda x: x.county_state_names.str.extract(
+                rf"(.+)_({state_pattern})"
+            )[0]
+        )
+        .assign(
+            state=lambda x: x.county_state_names.str.extract(
+                rf"(.+)_({state_pattern})"
+            )[1]
+        )
+        .rename(
+            columns={
+                "lat_county": "latitude",
+                "long_county": "longitude",
+                "fips": "county_id_fips",
+            }
+        )
+    )
+
+    return lat_long_fips
+
+
+def _add_time_cols(df: pd.DataFrame) -> pd.DataFrame:
+    """Add datetime and year_hour columns.
+
+    This function adds a datetime column and a year_hour column.
+    The datetime column is important for merging the data with other
+    data, and the year_hour 0-8769 column is important for modeling
+    purposes. The report_year column is also helpful for filtering,
+    so we keep all three!
+
+    For leap years (2020), December 31st is excluded.
+
+    TO-DO: decide whether to have the year_hour start at 0 or 1.
+    and update the date_range accordingly.
```

Comment:
Flagging this decision

Reply:
The data comes with an hour-of-year starting at 1, so I think we should probably stick with that convention.

Reply:
@aesharpe Can you explain why hour 1 is equal to YYYY-MM-DD 00:00 instead of 01:00? This isn't intuitive to me.

Reply:
This is tricky -- in `datetime` the year starts at `2022-01-01 00:00:00`. If we bumped it up by an hour and coorelated hour 1 with `2022-01-01 01:00:00` then the 8760th hour would be the next year: `2023-01-01 00:00:00 ` and not line up with the report year column which would still say 2022.

Reply:
But yeah, it's weird. I could see us going either way.